edge/pkg/devicetwin/dtclient/device_db.go

```diff
@@ -87,60 +88,80 @@ func UpdateDeviceMulti(updates []DeviceUpdate) error {
 
 //AddDeviceTrans the transaction of add device
 func AddDeviceTrans(adds []Device, addAttrs []DeviceAttr, addTwins []DeviceTwin) error {
-	var err error
-	obm := dbm.DBAccess
-	obm.Begin()
+	obm := dbm.DefaultOrmFunc()
+	err := obm.Begin()
+	if err != nil {
+		klog.Errorf("failed to begin transaction: %v", err)
+		return err
+	}
+
 	for _, add := range adds {
-		err = SaveDevice(&add)
+		err = SaveDevice(obm, &add)
 
 		if err != nil {
 			klog.Errorf("save device failed: %v", err)
-			obm.Rollback()
+			dbm.RollbackTransaction(obm)
```

Comment:
A defer function to run `dbm.RollbackTransaction(obm)` will be more clean up.

Reply:
Done.